internal/products.go

```diff
@@ -48,6 +48,39 @@ type Product struct {
 	Description  string       `json:"description"`
 }
 
+// Take the "Product" as returned from an RMT and adjust the repository URLs
+// to include a "credentials" parameter. This is somewhat specific to the RMT
+// instances available for Public Cloud on-demand instances, as they need to
+// authenticate to be able to access repositories.
+func fixRepoUrlsForRMT(p *Product) error {
+	for i := range p.Repositories {
```

Comment:
possible follow up: can we change the Repositories to be a slice of pointers? I think this way we can avoid the indexing.

Reply:
Yeah. I'll rework that as a followup PR.